[Android][ui] - Cannot pass rgb() values to DateTimePicker's elementColors object

### Minimal reproducible example

https://github.com/MariooC14/expo-ui-android-element-colors-bug-repro

### Steps to reproduce

For android: 
1. Create a project and install "@expo/ui".
2. Mount `DateTimePicker`, or any other component that accepts the `elementColors` prop. Pass in an object to the `elementProps` using an rgb value, e.g.:

```json
{
  "containerColor": "rgb(0,0,0)"
}
```

Error in console:
```
Cannot set prop 'elementColors' on view 'class expo.modules.kotlin.views.ComposeFunctionHolder'
→ Caused by: Cannot create a record of the type: 'expo.modules.ui.DateTimePickerColorOverrides'.
→ Caused by: Cannot cast 'String' for field 'containerColor' ('android.graphics.Color?').
→ Caused by: java.lang.IllegalArgumentException: Unknown color
```

Using a HEX value, however, produces expected results. 

Code I used to discover this:

```tsx
import { Host, DateTimePicker } from "@expo/ui/jetpack-compose";
import { DatePickerProps } from "./date-picker-types";
import { useTheme } from "react-native-paper";

export function DatePicker(props: DatePickerProps) {
  const theme = useTheme();
  console.log(
    "Rendering DatePicker with containerColor:",
    theme.colors.background,
  );

  return (
    <Host matchContents>
      <DateTimePicker
        onDateSelected={props.onDateSelected}
        displayedComponents="date"
        initialDate={props.initialDate.toISOString()}
        variant="picker"
        elementColors={{
          containerColor: theme.colors.background,
          // containerColor: "#1e1b16", // Equivalent but using Hex instead of rgb() - this version works
        }}
      />
    </Host>
  );
}
```

### Environment

```text
expo-env-info 2.0.11 environment info:
    System:
      OS: Windows 11 10.0.26200
    Binaries:
      Node: 22.14.0 - C:\dev\nodejs\node.EXE
      Yarn: 1.22.22 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 11.6.2 - C:\dev\nodejs\npm.CMD
    SDKs:
      Android SDK:
        API Levels: 34, 35, 36
        Build Tools: 35.0.0, 36.0.0, 36.1.0
        System Images: android-36.1 | Google Play Intel x86_64 Atom, android-36 | Google Play Intel x86_64 Atom, android-36 | Pre-Release 16 KB Page Size Google Play Intel x86_64 Atom
    npmPackages:
      expo: ~55.0.6 => 55.0.6
      react: 19.2.0 => 19.2.0
      react-native: 0.83.2 => 0.83.2
    Expo Workflow: bare
```

### Expo Doctor Diagnostics

```text
17/17 checks passed. No issues detected!
```

--- a/packages/expo-modules-core/CHANGELOG.md
+++ b/packages/expo-modules-core/CHANGELOG.md
@@ -12,6 +12,7 @@
 
 ### 🐛 Bug fixes
 
+- [Android] Added support for CSS `rgb(...)` and `rgba(...)` color strings when converting JS color values to `android.graphics.Color`. ([#44023](https://github.com/expo/expo/pull/44023) by [@dileepapeiris](https://github.com/dileepapeiris))
 - [Android] Fixed Compose view clipping that caused Material ripple effects to be cut off (e.g., Switch thumb ripple). ([#43656](https://github.com/expo/expo/pull/43656) by [@vonovak](https://github.com/vonovak))
 - [iOS] Fix memory leak due to retain cycle in SwiftUI views. ([#43468](https://github.com/expo/expo/pull/43468) by [@nishan](https://github.com/intergalacticspacehighway))
 - [iOS] Fixed compilation issues due to missing fallthrough case in `EXJavaScriptSerializable`. ([#43634](https://github.com/expo/expo/pull/43634) by [@tjzel](https://github.com/tjzel))

--- a/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/ColorTypeConverter.kt
+++ b/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/ColorTypeConverter.kt
@@ -14,6 +14,11 @@ import expo.modules.kotlin.jni.SingleType
 import androidx.core.graphics.toColorInt
 import com.facebook.react.bridge.ReadableArray
 
+private val rgbColorRegex = Regex(
+  pattern = """^rgba?\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*(?:,\s*([0-9]*\.?[0-9]+)\s*)?\)$""",
+  options = setOf(RegexOption.IGNORE_CASE)
+)
+
 /**
  * Color components for named colors following the [CSS3/SVG specification](https://www.w3.org/TR/css-color-3/#svg-color)
  * and additionally the transparent color.
@@ -211,7 +216,8 @@ class ColorTypeConverter : DynamicAwareTypeConverters<Color>() {
   }
 
   private fun colorFromString(value: String): Color {
-    val colorFromString = namedColors[value]
+    val normalizedValue = value.trim().lowercase()
+    val colorFromString = namedColors[normalizedValue]
     if (colorFromString != null) {
       return Color.valueOf(
         colorFromString[0],
@@ -221,7 +227,28 @@ class ColorTypeConverter : DynamicAwareTypeConverters<Color>() {
       )
     }
 
-    return Color.valueOf(value.toColorInt())
+    val rgbMatch = rgbColorRegex.matchEntire(normalizedValue)
+    if (rgbMatch != null) {
+      val red = rgbMatch.groupValues[1].toInt()
+      val green = rgbMatch.groupValues[2].toInt()
+      val blue = rgbMatch.groupValues[3].toInt()
+      require(red in 0..255 && green in 0..255 && blue in 0..255) {
+        "rgb() color component out of range"
+      }
+
+      val alphaGroup = rgbMatch.groupValues.getOrNull(4).orEmpty()
+      val alpha = if (alphaGroup.isNotEmpty()) {
+        val parsed = alphaGroup.toFloat()
+        require(parsed in 0f..1f) { "rgba() alpha out of range" }
+        parsed
+      } else {
+        1f
+      }
+
+      return Color.valueOf(red / 255f, green / 255f, blue / 255f, alpha)
+    }
+
+    return Color.valueOf(normalizedValue.toColorInt())
   }
 
   override fun getCppRequiredTypes(): ExpectedType =

--- a/packages/expo-modules-core/android/src/test/java/expo/modules/kotlin/types/ColorTypeConverterTest.kt
+++ b/packages/expo-modules-core/android/src/test/java/expo/modules/kotlin/types/ColorTypeConverterTest.kt
@@ -35,6 +35,43 @@ internal class ColorTypeConverterTest {
   }
 
   @Test
+  fun `converts from rgb string`() {
+    val colorString = DynamicFromObject("rgb(0,0,0)")
+
+    val color = convert<Color>(colorString)
+
+    Truth.assertThat(color.alpha()).isEqualTo(1f)
+    Truth.assertThat(color.red()).isEqualTo(0f)
+    Truth.assertThat(color.green()).isEqualTo(0f)
+    Truth.assertThat(color.blue()).isEqualTo(0f)
+  }
+
+  @Test
+  fun `converts from rgba string`() {
+    val colorString = DynamicFromObject("rgba(255, 0, 0, 0.5)")
+
+    val color = convert<Color>(colorString)
+
+    Truth.assertThat(color.alpha()).isEqualTo(0.5f)
+    Truth.assertThat(color.red()).isEqualTo(1f)
+    Truth.assertThat(color.green()).isEqualTo(0f)
+    Truth.assertThat(color.blue()).isEqualTo(0f)
+  }
+
+  @Test
+  fun `converts from rgb string with whitespace`() {
+    val colorString = DynamicFromObject(" rgb( 12 , 34 , 56 ) ")
+
+    val color = convert<Color>(colorString)
+
+    val expectedColor = arrayOf(12f, 34f, 56f).map { it / 255f }
+    Truth.assertThat(color.alpha()).isEqualTo(1f)
+    Truth.assertThat(color.red()).isEqualTo(expectedColor[0])
+    Truth.assertThat(color.green()).isEqualTo(expectedColor[1])
+    Truth.assertThat(color.blue()).isEqualTo(expectedColor[2])
+  }
+
+  @Test
   fun `converts from int`() {
     val colorInt = DynamicFromObject(Color.parseColor("#aabbcc").toDouble())
 

